### PR TITLE
Another day 3 in Go

### PR DESCRIPTION
**Pull request by tholok97**

### CODE REVIEW

File: days/day-03/solutions/day03_tholok97.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+// treeRow is an endlessly repeating pattern of trees.
+type treeRow struct {
+	// true == tree
+	// otherwise not tree
+	trees []bool
+}
+
+// is there a tree at position x, counting from left to right?
+func (r treeRow) tree(x int) bool {
+	x %= len(r.trees)
```
Comment (Arxcis) [inline days/day-03/solutions/day03_tholok97.go:18]: HAh. I have nerver seen `%=` before :laughing: :thinking: ... Cool. I have to think about this. What does it even mean to %=` something? :smile:

**Review by Arxcis — APPROVED** [review]
I learn something new everyday :smile: Approved!

File: days/day-03/solutions/day03_tholok97.go (hunk 1 above)
Comment (tholok97) [reply days/day-03/solutions/day03_tholok97.go:18]: TIL too, the linter suggested it to me as an alternative to `x = x % len(r.trees)` 🛩️

Comment (Arxcis) [reply days/day-03/solutions/day03_tholok97.go:18]: Thank you :+1: This is clear now :pray:

Comment (Arxcis) [reply days/day-03/solutions/day03_tholok97.go:18]: I don't like the linter's suggestion on this one. It is for some reason, very hard for me to read/understand. Maybe it is just me? :shrug:

Comment (tholok97) [reply days/day-03/solutions/day03_tholok97.go:18]: Hmmm this syntax is completely new to me, but it seems like this is the kind of situation it should be used for, and I assume the Go-peeps put it in there for a good reason, so I'm trying it on for size 👖

It's the same as `+=`, `-=` etc., but for `%` I guess

**Pull request merged**